### DISCUSSION TITLE
httpx   after  AsyncClient  ，Is there an example of how to get a cookie on the server side

### DISCUSSION THREAD

**Discussion by Ediczou497874** [encode/httpx#1481] [General]
httpx   after  AsyncClient  ，Is there an example of how to get a cookie on the server side？

**Comment by Ediczou497874** [comment]
class CwmpSession(object):
  """State machine to handle the lifecycle of a TCP session with the ACS."""

  def __init__(self,acs_url:str,username:str):
  
    self._httpclient= AsyncClient()
    self.cookies = self._httpclient._cookies

headers = [
            (
                b"Set-Cookie",
                b"1P_JAR=2020-08-09-18; expires=Tue, 08-Sep-2099 18:33:35 GMT; "
                b"path=/; domain=.baidu.com; Secure",
            ),
            (
                b"Set-Cookie",
                b"NID=204=KWdXOuypc86YvRfBSiWoW1dEXfSl_5qI7sxZY4umlk4J35yNTeNEkw15"
                b"MRaujK6uYCwkrtjihTTXZPp285z_xDOUzrdHt4dj0Z5C0VOpbvdLwRdHatHAzQs7"
                b"7TsaiWY78a3qU9r7KP_RbSLvLl2hlhnWFR2Hp5nWKPsAcOhQgSg; expires=Mon, "
                b"08-Feb-2099 18:33:35 GMT; path=/; domain=.baidu.com; HttpOnly",
            ),
        ]
        request = httpx.Request("GET", "https://www.baidu.com")
        response = httpx.Response(200, request=request)
       
        cookies = httpx.Cookies(jar)
       
        cookies.extract_cookies(response)
        t=cookies.get(name='BD_HOME')
        print(t)
        httpx.post()

**Reply by StephenBrown2** [reply]
Please surround your code with triple-ticks (` ``` `) and a language identifier (`python`) like so:

    ```python
    code
    ```

to allow easier reading and help. I tried to copy your code but it was not successful.

**Comment by Ediczou497874** [comment]
i  want   get the server cookie (eg   www.baidu.com)  , Best by  
resp = await self._httpclient.post(self.url,  auth=self.auth, data=data, **kwargs)。    

Looking forward to your reply

**Reply by StephenBrown2** [reply]
Searching the documentation for "cookies" brought me to https://www.python-httpx.org/quickstart/#cookies which details that "Any cookies that are set on the response can be easily accessed" which in your case may be gotten as:
```py
resp = await self._httpclient.post(self.url, auth=self.auth, data=data, **kwargs)
cookie = resp.cookies["BD_HOME"]
```
However I'm unsure if that answers your question.
